Sources/Diagnostics/DiagnosticsEvent.swift

```diff
@@ -15,13 +15,27 @@ import Foundation
 
 struct DiagnosticsEvent: Codable {
 
+    let id: UUID
     let version: Int = 1
     let eventType: DiagnosticsEvent.EventType
     let properties: [DiagnosticsPropertyKey: AnyEncodable]
     let timestamp: Date
+    let appSessionId: UUID?
```

Comment:
I'm guessing this is optional to make it backwards compatible with the existing cached.

If that's the case, if we're going to be discarding the existing cached events anyway due to #4843, then this can be nonoptional. Not a big deal though

Reply:
Yup that was the idea... but yeah you're right, I can make it non-nullable. It will be better 👍 Thank you! And yeah, I will hold merging this until after your PR. I think that should be a bit easier 😅